core/modules/filter/filter.pages.inc

```diff
@@ -50,15 +58,67 @@ function filter_format_editor_link_form($form, &$form_state, $format) {
 
   // Use a "textfield" rather than "url" to allow relative paths.
   $form['href'] = array(
-    '#title' => t('URL'),
+    '#title' => t('Link value'),
     '#type' => 'textfield',
     '#element_validate' => array('_filter_format_editor_link_url_validate'),
-    '#default_value' => isset($values['href']) ? $values['href'] : FALSE,
+    '#placeholder' => ' ',
+    '#default_value' => isset($values['href']) ? $values['href'] : NULL,
     '#parents' => array('attributes', 'href'),
+    '#wrapper_attributes' => array(
+      'data-editor-image-toggle' => t('Link to existing content'),
```

Comment:
@Graham-72 this is a good learning process. Since your work gets reviewed anyway, it's okay to make mistakes!

Reply:
Yes. thanks for your encouragement but I need to understand the terminology e.g. 'being namespaced under "image"'. 😃

Reply:
What he meant was "We should rename them, so that people don't think they are only used for toggling an image". 

I had assumed that's what the data attribute was for, so I had removed the data attribute in https://github.com/backdrop/backdrop/pull/2638 since toggling an image there didn't make any sense.

I can tell from this conversation that this data attribute is important, but I still have no idea why. I can't see any change in behavior with/without it on the `href` field.

Reply:
As I understand it the attribute data-editor-image-toggle is required for communicating from `filter.pages.inc` to the script `filter.js`. It gives the text to be used in the toggle action. Yes, it should have been renamed when extending the use of the toggle action to include inserting links as well as images.